statistics/statistics.go

```diff
@@ -21,329 +21,143 @@ package statistics
 import (
 	"errors"
 	"fmt"
-	"math"
-	"sort"
-	"time"
+	"strings"
 
-	log "github.com/Sirupsen/logrus"
 	"github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin"
-	"github.com/montanaflynn/stats"
 )
 
+type Plugin struct {
+	buffer map[string]*dataBuffer
+}
+
 const (
 	Name    = "statistics"
-	Version = 2
+	Version = 3
 )
 
-type Plugin struct {
-	buffer        map[string][]interface{}
-	bufferMaxSize int
-	bufferCurSize int
-	bufferIndex   int
-}
+var (
+	statList = []string{"count", "mean", "median", "standard_deviation", "variance", "95%_ile", "99%_ile", "2%_ile", "9%_ile", "25%_ile", "75%_ile", "91%_ile", "98%_ile", "minimum", "maximum", "range", "mode", "kurtosis", "skewness", "sum", "trimean", "first_quartile", "third_quartile", "quartile_range"}
```

Comment:
statList is different than Flag struct defined previously. Let's define them to be same and use Enum.